src/pip/_internal/operations/prepare.py

```diff
@@ -459,32 +458,43 @@ def _get_linked_req_hashes(self, req):
         # showing the user what the hash should be.
         return req.hashes(trust_internet=False) or MissingHashes()
 
-    def _fetch_metadata(preparer, link):
+    def _fetch_metadata_using_lazy_wheel(self, link):
         # type: (Link) -> Optional[Distribution]
-        """Fetch metadata, using lazy wheel if possible."""
-        use_lazy_wheel = preparer.use_lazy_wheel
-        remote_wheel = link.is_wheel and not link.is_file
-        if use_lazy_wheel and remote_wheel and not preparer.require_hashes:
-            wheel = Wheel(link.filename)
-            name = canonicalize_name(wheel.name)
-            # If HTTPRangeRequestUnsupported is raised, fallback silently.
-            with indent_log(), suppress(HTTPRangeRequestUnsupported):
-                logger.info(
-                    'Obtaining dependency information from %s %s',
-                    name, wheel.version,
-                )
-                url = link.url.split('#', 1)[0]
-                session = preparer.downloader.session
-                return dist_from_wheel_url(name, url, session)
-        return None
+        """Fetch metadata using lazy wheel, if possible."""
+        if not self.use_lazy_wheel:
+            return None
+        if self.require_hashes:
+            logger.debug('Lazy wheel is not used as hash checking is required')
+            return None
+        if link.is_file or not link.is_wheel:
+            logger.debug(
+                'Lazy wheel is not used as '
+                '%r does not points to a remote wheel',
+                link,
+            )
+            return None
+
+        wheel = Wheel(link.filename)
+        name = canonicalize_name(wheel.name)
+        logger.info(
+            'Obtaining dependency information from %s %s',
+            name, wheel.version,
+        )
+        url, sep, checksum = link.url.partition('#')
```

Comment:
We shouldn't introduce local variables that we don't use.

Reply:
I will revert this back to the `str.split` one—I don't like how it look, but this is probably worse :smile: